From 515110c1fc56b998d3f90c080676bef77f1a3803 Mon Sep 17 00:00:00 2001
From: ninja- <xninja@openmailbox.org>
Date: Tue, 8 Oct 2013 14:35:58 +0200
Subject: [PATCH] Add respawn API.


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 610a97c9..2cd40fcf 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1328,6 +1328,13 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
         public void setCollidesWithEntities(boolean collides) {
             throw new UnsupportedOperationException("Not supported yet.");
         }
+
+        /**
+         * Respawns the player if dead.
+         */
+        public void respawn() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
     }
 
     @NotNull
-- 
2.25.1

