From 318e0258e57da182ff7e305b70136d91c6499e90 Mon Sep 17 00:00:00 2001
From: drXor <mcyoungsota@gmail.com>
Date: Sun, 23 Feb 2014 16:16:29 -0400
Subject: [PATCH] Silenceable Lightning API


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index beb9d8f2..2627e5d3 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -2081,6 +2081,31 @@ public interface World extends PluginMessageRecipient, Metadatable {
     public class Spigot
     {
 
+        /**
+         * Strikes lightning at the given {@link Location} and possibly without sound
+         *
+         * @param loc The location to strike lightning
+         * @param isSilent Whether this strike makes no sound
+         * @return The lightning entity.
+         */
+        @NotNull
+        public LightningStrike strikeLightning(@NotNull Location loc, boolean isSilent)
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
+
+        /**
+         * Strikes lightning at the given {@link Location} without doing damage and possibly without sound
+         *
+         * @param loc The location to strike lightning
+         * @param isSilent Whether this strike makes no sound
+         * @return The lightning entity.
+         */
+        @NotNull
+        public LightningStrike strikeLightningEffect(@NotNull Location loc, boolean isSilent)
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
     }
 
     @NotNull
diff --git a/src/main/java/org/bukkit/entity/LightningStrike.java b/src/main/java/org/bukkit/entity/LightningStrike.java
index 39ceb444..5904ed87 100644
--- a/src/main/java/org/bukkit/entity/LightningStrike.java
+++ b/src/main/java/org/bukkit/entity/LightningStrike.java
@@ -18,6 +18,15 @@ public interface LightningStrike extends Entity {
     public class Spigot extends Entity.Spigot
     {
 
+        /*
+         * Returns whether the strike is silent.
+         *
+         * @return whether the strike is silent.
+         */
+        public boolean isSilent()
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
     }
 
     @NotNull
-- 
2.20.1

