From 52655a219fa9b7046a3c36185424fed1ee3f31fc Mon Sep 17 00:00:00 2001
From: ninja- <xninja@openmailbox.org>
Date: Tue, 8 Oct 2013 14:35:58 +0200
Subject: [PATCH] Add respawn API.


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 856d86d1..b8ecfaef 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1533,6 +1533,14 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
         {
             throw new UnsupportedOperationException( "Not supported yet." );
         }
+
+        /**
+         * Respawns the player if dead.
+         */
+        public void respawn()
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
     }
 
     @Override
-- 
2.19.1

