From 66ef492d4b4fa44bcc21d5e66747aa79f2ab13bc Mon Sep 17 00:00:00 2001
From: libraryaddict <redwarfare@live.com>
Date: Fri, 22 Aug 2014 05:31:04 -0400
Subject: [PATCH] Added isUnbreakable and setUnbreakable to ItemMeta


diff --git a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
index 0bc410ae..4e8c7a8a 100644
--- a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
+++ b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
@@ -393,6 +393,29 @@ public interface ItemMeta extends Cloneable, ConfigurationSerializable, Persiste
     public class Spigot
     {
 
+        /**
+         * Sets the unbreakable tag
+         *
+         * @param unbreakable true if set unbreakable
+         * @deprecated see {@link ItemMeta#setUnbreakable(boolean)}
+         */
+        @Deprecated
+        public void setUnbreakable(boolean unbreakable)
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
+
+        /**
+         * Return if the unbreakable tag is true
+         *
+         * @return true if the unbreakable tag is true
+         * @deprecated see {@link ItemMeta#isUnbreakable()}
+         */
+        @Deprecated
+        public boolean isUnbreakable()
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
     }
 
     @NotNull
-- 
2.20.1

