From 8261b9c0fbe9a6278e6697fb02fac2e9ba0efa26 Mon Sep 17 00:00:00 2001
From: libraryaddict <redwarfare@live.com>
Date: Fri, 22 Aug 2014 05:31:04 -0400
Subject: [PATCH] Added isUnbreakable and setUnbreakable to ItemMeta


diff --git a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
index e45de7a4..f2a6755a 100644
--- a/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
+++ b/src/main/java/org/bukkit/inventory/meta/ItemMeta.java
@@ -348,6 +348,29 @@ public interface ItemMeta extends Cloneable, ConfigurationSerializable {
     public class Spigot
     {
 
+        /**
+         * Sets the unbreakable tag
+         *
+         * @param unbreakable true if set unbreakable
+         * @deprecated see {@link ItemMeta#setUnbreakable(boolean)}
+         */
+        @Deprecated
+        public void setUnbreakable(boolean unbreakable)
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
+
+        /**
+         * Return if the unbreakable tag is true
+         *
+         * @return true if the unbreakable tag is true
+         * @deprecated see {@link ItemMeta#isUnbreakable()}
+         */
+        @Deprecated
+        public boolean isUnbreakable()
+        {
+            throw new UnsupportedOperationException( "Not supported yet." );
+        }
     }
 
     @NotNull
-- 
2.19.1

