From 3b0e70d84c04706a07ad241b38a6a5003ccb0a92 Mon Sep 17 00:00:00 2001
From: md_5 <md_5@live.com.au>
Date: Sun, 2 Jun 2013 15:16:05 +1000
Subject: [PATCH] Implement Arrow API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftArrow.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftArrow.java
index 78de5ea9..d9559286 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftArrow.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftArrow.java
@@ -110,6 +110,17 @@ public class CraftArrow extends AbstractProjectile implements Arrow {
     // Spigot start
     private final Arrow.Spigot spigot = new Arrow.Spigot()
     {
+        @Override
+        public double getDamage()
+        {
+            return getHandle().getDamage();
+        }
+
+        @Override
+        public void setDamage(double damage)
+        {
+            getHandle().setDamage( damage );
+        }
     };
 
     public Arrow.Spigot spigot()
-- 
2.20.1

