From a01879588f12bfca78816ee0783124a3836b5c6a Mon Sep 17 00:00:00 2001
From: md_5 <md_5@live.com.au>
Date: Thu, 27 Jun 2013 17:26:09 +1000
Subject: [PATCH] Properly Close Inventories

Properly close inventories when unloading and switching worlds.

diff --git a/src/main/java/net/minecraft/server/level/WorldServer.java b/src/main/java/net/minecraft/server/level/WorldServer.java
index 317519ea6..70d0d321d 100644
--- a/src/main/java/net/minecraft/server/level/WorldServer.java
+++ b/src/main/java/net/minecraft/server/level/WorldServer.java
@@ -1033,6 +1033,15 @@ public class WorldServer extends World implements GeneratorAccessSeed {
     }
 
     public void unloadChunk(Chunk chunk) {
+        // Spigot Start
+        for (TileEntity tileentity : chunk.getTileEntities().values()) {
+            if (tileentity instanceof net.minecraft.world.IInventory) {
+                for (org.bukkit.entity.HumanEntity h : Lists.newArrayList(((net.minecraft.world.IInventory) tileentity).getViewers())) {
+                    h.closeInventory();
+                }
+            }
+        }
+        // Spigot End
         chunk.C();
     }
 
@@ -1872,6 +1881,13 @@ public class WorldServer extends World implements GeneratorAccessSeed {
 
         public void a(Entity entity) {
             org.spigotmc.AsyncCatcher.catchOp("entity unregister"); // Spigot
+            // Spigot Start
+            if (entity.getBukkitEntity() instanceof org.bukkit.inventory.InventoryHolder) {
+                for (org.bukkit.entity.HumanEntity h : Lists.newArrayList(((org.bukkit.inventory.InventoryHolder) entity.getBukkitEntity()).getInventory().getViewers())) {
+                    h.closeInventory();
+                }
+            }
+            // Spigot End
             WorldServer.this.getChunkProvider().removeEntity(entity);
             if (entity instanceof EntityPlayer) {
                 EntityPlayer entityplayer = (EntityPlayer) entity;
-- 
2.25.1

