From aa213277ef4f6ca2afcb43023237c02d296fa2b9 Mon Sep 17 00:00:00 2001
From: md_5 <md_5@live.com.au>
Date: Fri, 28 Jun 2013 19:52:54 +1000
Subject: [PATCH] Disallow Interaction With Self


diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index db27bb1fd..8cb47755b 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -2026,6 +2026,13 @@ public class PlayerConnection implements ServerPlayerConnection, PacketListenerP
         if (this.player.isImmobile()) return; // CraftBukkit
         WorldServer worldserver = this.player.getLevel();
         final Entity entity = packetplayinuseentity.getTarget(worldserver);
+        // Spigot Start
+        if ( entity == player && !player.isSpectator() )
+        {
+            disconnect( "Cannot interact with self!" );
+            return;
+        }
+        // Spigot End
 
         this.player.resetLastActionTime();
         this.player.setShiftKeyDown(packetplayinuseentity.isUsingSecondaryAction());
-- 
2.25.1

