From 7580f488d1d4c626174447f8eca6e144696cccca Mon Sep 17 00:00:00 2001
From: md_5 <md_5@live.com.au>
Date: Thu, 5 Dec 2013 13:55:53 +1100
Subject: [PATCH] Fix Broken Async Chat


diff --git a/src/main/java/net/minecraft/network/protocol/game/PacketPlayInChat.java b/src/main/java/net/minecraft/network/protocol/game/PacketPlayInChat.java
index 7e63ce9e7..1a03b59a1 100644
--- a/src/main/java/net/minecraft/network/protocol/game/PacketPlayInChat.java
+++ b/src/main/java/net/minecraft/network/protocol/game/PacketPlayInChat.java
@@ -25,7 +25,24 @@ public class PacketPlayInChat implements Packet<PacketListenerPlayIn> {
         packetdataserializer.a(this.message);
     }
 
-    public void a(PacketListenerPlayIn packetlistenerplayin) {
+    // Spigot Start
+    private static final java.util.concurrent.ExecutorService executors = java.util.concurrent.Executors.newCachedThreadPool(
+            new com.google.common.util.concurrent.ThreadFactoryBuilder().setDaemon( true ).setNameFormat( "Async Chat Thread - #%d" ).build() );
+    public void a(final PacketListenerPlayIn packetlistenerplayin) {
+        if ( !message.startsWith("/") )
+        {
+            executors.submit( new Runnable()
+            {
+
+                @Override
+                public void run()
+                {
+                    packetlistenerplayin.a( PacketPlayInChat.this );
+                }
+            } );
+            return;
+        }
+        // Spigot End
         packetlistenerplayin.a(this);
     }
 
-- 
2.25.1

