From 3df8bc06b9937d756bc453e9a48e2b74afdf940e Mon Sep 17 00:00:00 2001
From: agentk20 <agentkid20@gmail.com>
Date: Sat, 3 Aug 2013 19:28:48 +1000
Subject: [PATCH] Fully Disable Snooper When Not Required


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 69e14dfc9..942a9757e 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -815,11 +815,11 @@ public abstract class MinecraftServer implements IAsyncTaskHandler, IMojangStati
         }
 
         this.methodProfiler.enter("snooper");
-        if (!this.snooper.d() && this.ticks > 100) {
+        if (getSnooperEnabled() && !this.snooper.d() && this.ticks > 100) { // Spigot
             this.snooper.a();
         }
 
-        if (this.ticks % 6000 == 0) {
+        if (getSnooperEnabled() && this.ticks % 6000 == 0) { // Spigot
             this.snooper.b();
         }
 
-- 
2.19.1

