From bf1598769fd78f3a63cdbaa76f2b85b6f542c1d6 Mon Sep 17 00:00:00 2001
From: ninja- <xninja@openmailbox.org>
Date: Tue, 8 Oct 2013 14:34:49 +0200
Subject: [PATCH] Implement respawn API.


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index d93b9f65..f81d3884 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1461,6 +1461,15 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         public void setCollidesWithEntities(boolean collides) {
             CraftPlayer.this.setCollidable(collides);
         }
+
+        @Override
+        public void respawn()
+        {
+            if ( getHealth() <= 0 && isOnline() )
+            {
+                server.getServer().getPlayerList().moveToWorld( getHandle(), 0, false );
+            }
+        }
     };
 
     public Player.Spigot spigot()
-- 
2.11.0

