From 5ed8d5a5e3e8828c368de8fbce03c817e982ce0e Mon Sep 17 00:00:00 2001
From: Smove <jan@lavasurvival.net>
Date: Sat, 1 Feb 2014 18:12:16 +1100
Subject: [PATCH] Implement Locale Getter for Players


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 98a346e9f..f262634d9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1535,6 +1535,12 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
                 getHandle().playerConnection.sendPacket( packet );
             }
         }
+
+        @Override
+        public String getLocale()
+        {
+           return getHandle().locale;
+        }
     };
 
     public Player.Spigot spigot()
-- 
2.11.0

