From a437c0d765f56d9a5298b9c2c77a436457c1638d Mon Sep 17 00:00:00 2001
From: Thinkofdeath <thethinkofdeath@gmail.com>
Date: Tue, 14 Jan 2014 20:11:25 +0000
Subject: [PATCH] Fix ConcurrentModificationException while being idle kicked
 in a vehicle


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 03c1a72..3174a71 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1,4 +1,4 @@
-package net.minecraft.server;
+	package net.minecraft.server;
 
 import com.google.common.base.Function;
 import com.google.common.base.Objects;
@@ -1056,6 +1056,7 @@ public abstract class World implements IBlockAccess {
             this.everyoneSleeping();
         }
 
+        if (!guardEntityList) { // Spigot - It will get removed after the tick if we are ticking
         int i = entity.ac;
         int j = entity.ae;
 
@@ -1072,6 +1073,7 @@ public abstract class World implements IBlockAccess {
             this.entityList.remove(index);
         }
         // CraftBukkit end
+        } // Spigot
         this.c(entity);
     }
 
-- 
2.7.4

