From 36f6d7ae71e961a5336872298e52a4bff5c10150 Mon Sep 17 00:00:00 2001
From: Thinkofdeath <thethinkofdeath@gmail.com>
Date: Tue, 14 Jan 2014 20:11:25 +0000
Subject: [PATCH] Fix ConcurrentModificationException while being idle kicked
 in a vehicle


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 854a545..bba856d 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1,4 +1,4 @@
-package net.minecraft.server;
+	package net.minecraft.server;
 
 import com.google.common.base.Function;
 import com.google.common.base.Objects;
@@ -1043,6 +1043,7 @@ public abstract class World implements IBlockAccess {
             this.everyoneSleeping();
         }
 
+        if (!guardEntityList) { // Spigot - It will get removed after the tick if we are ticking
         int i = entity.ac;
         int j = entity.ae;
 
@@ -1059,6 +1060,7 @@ public abstract class World implements IBlockAccess {
             this.entityList.remove(index);
         }
         // CraftBukkit end
+        } // Spigot
         this.c(entity);
     }
 
-- 
2.7.4

