From db91fcdd426addeb03a60e04674cb959eccf9bd6 Mon Sep 17 00:00:00 2001
From: drXor <mcyoungsota@gmail.com>
Date: Thu, 17 Jul 2014 21:13:15 -0400
Subject: [PATCH] Fix for enchanting table wiping meta when placing stacked
 items


diff --git a/src/main/java/net/minecraft/server/ContainerEnchantTable.java b/src/main/java/net/minecraft/server/ContainerEnchantTable.java
index ef9c7b296..5dce765b2 100644
--- a/src/main/java/net/minecraft/server/ContainerEnchantTable.java
+++ b/src/main/java/net/minecraft/server/ContainerEnchantTable.java
@@ -369,7 +369,11 @@ public class ContainerEnchantTable extends Container {
                     ((Slot) this.slots.get(0)).set(itemstack1.cloneItemStack());
                     itemstack1.setCount(0);
                 } else if (!itemstack1.isEmpty()) {
-                    ((Slot) this.slots.get(0)).set(new ItemStack(itemstack1.getItem()));
+                    // Spigot start
+                    ItemStack clone = itemstack1.cloneItemStack();
+                    clone.setCount(1);
+                    ((Slot) this.slots.get(0)).set(clone);
+                    // Spigot end
                     itemstack1.subtract(1);
                 }
             }
-- 
2.17.1

