From 07aab9e251bd6959e0680c02026fa8bd921e89e6 Mon Sep 17 00:00:00 2001
From: Thinkofdeath <thinkofdeath@spigotmc.org>
Date: Sat, 19 Jul 2014 12:49:48 +0100
Subject: [PATCH] Don't spawn bonus ocelots when plugins spawn ocelots


diff --git a/src/main/java/net/minecraft/server/EntityOcelot.java b/src/main/java/net/minecraft/server/EntityOcelot.java
index 18171d7d0..22bbe2c8e 100644
--- a/src/main/java/net/minecraft/server/EntityOcelot.java
+++ b/src/main/java/net/minecraft/server/EntityOcelot.java
@@ -10,6 +10,7 @@ public class EntityOcelot extends EntityTameableAnimal {
     private static final MinecraftKey bI = new MinecraftKey("cat");
     private PathfinderGoalAvoidTarget<EntityHuman> bJ;
     private PathfinderGoalTempt bK;
+    public boolean spawnBonus = true; // Spigot
 
     public EntityOcelot(World world) {
         super(EntityTypes.OCELOT, world);
@@ -234,7 +235,7 @@ public class EntityOcelot extends EntityTameableAnimal {
     @Nullable
     public GroupDataEntity prepare(DifficultyDamageScaler difficultydamagescaler, @Nullable GroupDataEntity groupdataentity, @Nullable NBTTagCompound nbttagcompound) {
         groupdataentity = super.prepare(difficultydamagescaler, groupdataentity, nbttagcompound);
-        if (this.getCatType() == 0 && this.world.random.nextInt(7) == 0) {
+        if (spawnBonus && this.getCatType() == 0 && this.world.random.nextInt(7) == 0) { // Spigot
             for (int i = 0; i < 2; ++i) {
                 EntityOcelot entityocelot = new EntityOcelot(this.world);
 
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 44f8127c6..ee8835a8d 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -1206,6 +1206,12 @@ public class CraftWorld implements World {
         }
 
         if (entity != null) {
+            // Spigot start
+            if (entity instanceof EntityOcelot)
+            {
+                ( (EntityOcelot) entity ).spawnBonus = false;
+            }
+            // Spigot end
             return entity;
         }
 
-- 
2.17.1

