From ddbfbbe493f58865b1a9c4ca19d8bcfc0ecd2bd4 Mon Sep 17 00:00:00 2001
From: "Evan A. Haskell" <eah2119@gmail.com>
Date: Thu, 26 Jun 2014 18:37:29 -0400
Subject: [PATCH] Fix Null Tile Entity Worlds


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index a2c966db2..b34d6007a 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1734,6 +1734,7 @@ public abstract class World implements IEntityAccess, GeneratorAccess, IIBlockAc
                         }
                     }
 
+                    tileentity.setWorld(this); // Spigot - No null worlds
                     this.c.add(tileentity);
                 } else {
                     this.getChunkAtWorldCoords(blockposition).a(blockposition, tileentity);
-- 
2.17.1

