From 4e9b1ff7edb68651e8c1dffbf3af9b32300016e5 Mon Sep 17 00:00:00 2001
From: "Evan A. Haskell" <eah2119@gmail.com>
Date: Thu, 26 Jun 2014 18:37:29 -0400
Subject: [PATCH] Fix Null Tile Entity Worlds


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index d772d0e8a..3df194528 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1724,6 +1724,7 @@ public abstract class World implements GeneratorAccess, IIBlockAccess, AutoClose
                         }
                     }
 
+                    tileentity.setWorld(this); // Spigot - No null worlds
                     this.c.add(tileentity);
                 } else {
                     this.getChunkAtWorldCoords(blockposition).a(blockposition, tileentity);
-- 
2.17.1

