From 2d4ce120c8b1a7fb3c504d235d31bcf59b7ca616 Mon Sep 17 00:00:00 2001
From: libraryaddict <redwarfare@live.com>
Date: Fri, 22 Aug 2014 05:35:16 -0400
Subject: [PATCH] Added isUnbreakable and setUnbreakable to ItemMeta


diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
index 243d4354f..99a2c8c8b 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
@@ -1297,6 +1297,17 @@ class CraftMetaItem implements ItemMeta, Damageable, Repairable {
     // Spigot start
     private final Spigot spigot = new Spigot()
     {
+        @Override
+        public void setUnbreakable(boolean setUnbreakable)
+        {
+            CraftMetaItem.this.setUnbreakable(setUnbreakable);
+        }
+
+        @Override
+        public boolean isUnbreakable()
+        {
+            return CraftMetaItem.this.unbreakable;
+        }
     };
 
     @Override
-- 
2.19.1

