From 6f01bcaa8b5505cc1cbcf4c7b431e8c8d105063a Mon Sep 17 00:00:00 2001
From: BlackHole <black-hole@live.com>
Date: Fri, 22 Apr 2016 16:43:47 +0200
Subject: [PATCH] Don't tick blocks if neighbor chunks aren't loaded


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 10a335196..23c80f80a 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -485,6 +485,7 @@ public class WorldServer extends World implements IAsyncTaskHandler {
                 chunk.x();
                 this.methodProfiler.exitEnter("tickChunk");
                 chunk.d(false);
+                if ( !chunk.areNeighborsLoaded( 1 ) ) continue; // Spigot
                 this.methodProfiler.exitEnter("thunder");
                 int l;
                 BlockPosition blockposition;
-- 
2.19.1

