From 9253d0ab07068d8909b28f6bdfa8d8a2f39e964c Mon Sep 17 00:00:00 2001
From: "Evan A. Haskell" <eah2119@gmail.com>
Date: Thu, 26 Jun 2014 18:37:29 -0400
Subject: [PATCH] Fix Null Tile Entity Worlds


diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 7cbd2e7..24ef2b0 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1976,6 +1976,7 @@ public abstract class World implements IBlockAccess {
                         }
                     }
 
+                    tileentity.a(this); // Spigot - No null worlds
                     this.b.add(tileentity);
                 } else {
                     this.a(tileentity);
-- 
2.7.4

