From 001d4babee37d35db88218cfea14eebce1e1aa3c Mon Sep 17 00:00:00 2001
From: md_5 <git@md-5.net>
Date: Thu, 11 Jan 2018 08:23:43 +1100
Subject: [PATCH] Allow changing minimum squid spawn range


diff --git a/src/main/java/net/minecraft/server/EntitySquid.java b/src/main/java/net/minecraft/server/EntitySquid.java
index 16fabe5ff..5d7d0b691 100644
--- a/src/main/java/net/minecraft/server/EntitySquid.java
+++ b/src/main/java/net/minecraft/server/EntitySquid.java
@@ -167,7 +167,7 @@ public class EntitySquid extends EntityWaterAnimal {
     }
 
     public boolean a(GeneratorAccess generatoraccess, boolean flag) {
-        return this.locY > 45.0D && this.locY < (double) generatoraccess.getSeaLevel();
+        return this.locY > this.world.spigotConfig.squidSpawnRangeMin && this.locY < (double) generatoraccess.getSeaLevel(); // Spigot
     }
 
     public void c(float f, float f1, float f2) {
diff --git a/src/main/java/org/spigotmc/SpigotWorldConfig.java b/src/main/java/org/spigotmc/SpigotWorldConfig.java
index e9c68cf13..95f7a9587 100644
--- a/src/main/java/org/spigotmc/SpigotWorldConfig.java
+++ b/src/main/java/org/spigotmc/SpigotWorldConfig.java
@@ -306,4 +306,10 @@ public class SpigotWorldConfig
         entityMaxTickTime = getInt("max-tick-time.entity", 50);
         log("Tile Max Tick Time: " + tileMaxTickTime + "ms Entity max Tick Time: " + entityMaxTickTime + "ms");
     }
+
+    public double squidSpawnRangeMin;
+    private void squidSpawnRange()
+    {
+        squidSpawnRangeMin = getDouble("squid-spawn-range.min", 45.0D);
+    }
 }
-- 
2.17.1

