From 8029ae3d86cc67124953a697c4cc8f4943336c2c Mon Sep 17 00:00:00 2001
From: md_5 <git@md-5.net>
Date: Thu, 11 Jan 2018 08:23:43 +1100
Subject: [PATCH] Allow changing minimum squid spawn range


diff --git a/src/main/java/net/minecraft/server/EntitySquid.java b/src/main/java/net/minecraft/server/EntitySquid.java
index b1e8924ed..0ce16be65 100644
--- a/src/main/java/net/minecraft/server/EntitySquid.java
+++ b/src/main/java/net/minecraft/server/EntitySquid.java
@@ -141,7 +141,7 @@ public class EntitySquid extends EntityWaterAnimal {
     }
 
     public boolean P() {
-        return this.locY > 45.0D && this.locY < (double) this.world.getSeaLevel() && super.P();
+        return this.locY > this.world.spigotConfig.squidSpawnRangeMin && this.locY < (double) this.world.getSeaLevel() && super.P(); // Spigot
     }
 
     public void b(float f, float f1, float f2) {
diff --git a/src/main/java/org/spigotmc/SpigotWorldConfig.java b/src/main/java/org/spigotmc/SpigotWorldConfig.java
index 01e73eb89..2eea84bab 100644
--- a/src/main/java/org/spigotmc/SpigotWorldConfig.java
+++ b/src/main/java/org/spigotmc/SpigotWorldConfig.java
@@ -306,4 +306,10 @@ public class SpigotWorldConfig
         entityMaxTickTime = getInt("max-tick-time.entity", 50);
         log("Tile Max Tick Time: " + tileMaxTickTime + "ms Entity max Tick Time: " + entityMaxTickTime + "ms");
     }
+
+    public double squidSpawnRangeMin;
+    private void squidSpawnRange()
+    {
+        squidSpawnRangeMin = getDouble("squid-spawn-range.min", 45.0D);
+    }
 }
-- 
2.14.1

