From c7d35284bba2eb5728594ce32606227075ed8be0 Mon Sep 17 00:00:00 2001
From: BlackHole <black-hole@live.com>
Date: Fri, 22 Apr 2016 16:43:47 +0200
Subject: [PATCH] Don't tick blocks if neighbor chunks aren't loaded


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 88938425..391a09c5 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -423,6 +423,7 @@ public class WorldServer extends World implements IAsyncTaskHandler {
                 chunk.n();
                 this.methodProfiler.c("tickChunk");
                 chunk.b(false);
+                if ( !chunk.areNeighborsLoaded( 1 ) ) continue; // Spigot
                 this.methodProfiler.c("thunder");
                 int l;
                 BlockPosition blockposition;
-- 
2.11.0

